patches/chrome-browser-resources-settings-people_page-import_data_dialog.js.patch

```diff
@@ -15,7 +15,9 @@ index 4192db5cdfb0ff0600460ac20b7151e4794a9d40..94033e6abd64c1cefdc7ae9a721f8bda
 +        !(this.getPref('import_dialog_ledger').value &&
 +          this.selected_.ledger) &&
 +        !(this.getPref('import_dialog_windows').value &&
-+          this.selected_.windows);
++          this.selected_.windows) &&
++        !(this.getPref('import_dialog_extensions').value &&
++          this.selected_.extensions);
```

Comment:
@petemill please advice if you have any idea about avoiding this patch.

Reply:
I'm fairly certain there's a better way to do this

Reply:
Yes I'm trying to see if we can override this via behavior

Reply:
I think we can remove this whole patch and extend `prefsChanged_` via a behavior. 

In brave_settings_overrides:
```js
BravePatching.RegisterPolymerComponentBehaviors({
  'settings-import-data-dialog': [{
  registered: function () {
    const oldPrefsChanged = this.prefsChanged_
    if (!oldPrefsChanged) {
      console.error('[Brave Settings Overrides] cannot find prefsChanged_ on ImportDataDialog')
      return
    }
    this.prefsChanged_ = function () {
      if (typeof this.noImportDataTypeSelected_ !== 'boolean') {
        console.error('[Brave Settings Overrides] cannot find noImportDataTypeSelected_ on ImportDataDialog')
        return
      }
      oldPrefsChanged.apply(this)
      if (this.selected_ == undefined || this.prefs == undefined) {
        return;
      }
      this.noImportDataTypeSelected_ = this.noImportDataTypeSelected_ &&
        !(this.getPref('import_dialog_cookies').value &&
        this.selected_.cookies) &&
        !(this.getPref('import_dialog_stats').value &&
          this.selected_.stats) &&
        !(this.getPref('import_dialog_ledger').value &&
          this.selected_.ledger) &&
        !(this.getPref('import_dialog_windows').value &&
          this.selected_.windows) &&
        !(this.getPref('import_dialog_extensions').value &&
          this.selected_.extensions)
    }
  }
}]
})
```

Another option would be to instead override `shouldDisableImport_` which would prevent having to copy the early return, but that changes the logic from a push to a pull, and I'm not sure if the notification will work.

Reply:
@petemill Thanks! this works well. :thu